samples/template-operator/README.md

```diff
@@ -36,117 +36,206 @@ MODULE_NAME ?= template
+
+### Steps:
+
+1. Refer to [API definition](./operator/api/v1alpha1/sample_types.go) of `SampleCR` and implement `Status` sub-resource similarly.
+   This `Status` type definition is sourced from the `module-manager` declarative library and contains all valid `.status.state` values as discussed in the previous sections.
+   ```yaml
+    Status types.Status `json:"status,omitempty"`
+   ```
+   
+2. Ensure the module CR implements the `module-manager` declarative library's resource requirements, represented by `types.CustomObject`. Also implement missing interface methods.
+   ```go
+   var _ types.CustomObject = &Sample{}
+
+   func (s *Sample) GetStatus() types.Status {
+        return s.Status
+   }
+   
+   func (s *Sample) SetStatus(status types.Status) {
+        s.Status = status
+   }
+   
+   func (s *Sample) ComponentName() string {
+        return "sample-component-name"
+   }
+   ```
+
+3. Refer to the [controller implementation](./operator/controllers/sample_controller.go). 
+Instead of implementing the default reconciler interface, as provided by `kuberbuilder`, include the `module-manager` declarative reconciler.
+   ```go
+   // SampleReconciler reconciles a Sample object
+   type SampleReconciler struct {
+        declarative.ManifestReconciler
+        client.Client
+        Scheme *runtime.Scheme
+        *rest.Config
+   }
+   ```
+   Notice there is not `Reconcile()` method implemented in this controller, since the logic is abstracted within the declarative reconciler.
+   
+4. As part of reconciler's SetupWithManager() in the Sample CR [controller implementation](./operator/controllers/sample_controller.go), declarative options have been used.
+   ```go
+   return r.Inject(mgr, &v1alpha1.Sample{},
+        declarative.WithManifestResolver(manifestResolver),
+        declarative.WithResourceLabels(map[string]string{"sampleKey": "sampleValue"}),
+        declarative.WithObjectTransform(transform),
+        declarative.WithResourcesReady(true), 
+   )
+   ```
+   These options can be used modify manifest installation and uninstallation. Some options are applied as a manifest pre-processing step and others as post-processing.
+   More details on these steps can be found in the [options documentation](https://github.com/kyma-project/module-manager/blob/main/operator/pkg/declarative/options.go).
+
+5. A manadatory requirement of this reconciler is to provide the option `declarative.WithManifestResolver(manifestResolver)`, as it holds the chart information to be processed by the declarative reconciler. 
+
+   This ManifestResolver should implement `types.ManifestResolver` from the declarative library. 
+   E.g. Sample CR [controller implementation](./operator/controllers/sample_controller.go) returns chart information.
+   ```go
+      // Get returns the chart information to be processed.
+	  func (m *ManifestResolver) Get(obj types.BaseCustomObject) (types.InstallationSpec, error) {
```

Comment:
```suggestion
	  func (m *ManifestResolver) Get(obj runtime.Object) (types.InstallationSpec, error) {
```
IMO we should not deviate from runtime here.

Reply:
Let's discuss this..